src/knx/application_program_object.cpp

```diff
@@ -11,7 +11,16 @@ ApplicationProgramObject::ApplicationProgramObject(Memory& memory)
     Property* properties[] =
     {
         new DataProperty(PID_OBJECT_TYPE, false, PDT_UNSIGNED_INT, 1, ReadLv3 | WriteLv0, (uint16_t)OT_APPLICATION_PROG),
-        new DataProperty(PID_PROG_VERSION, true, PDT_GENERIC_05, 1, ReadLv3 | WriteLv3),
+        new CallbackProperty<ApplicationProgramObject>(this, PID_PROG_VERSION, true, PDT_GENERIC_05, 1, ReadLv3 | WriteLv3,
+            [](ApplicationProgramObject* io, uint16_t start, uint8_t count, uint8_t* data) -> uint8_t {
+                pushByteArray(io->_programVersion, 5, data);
+                return 1;
+            },
+            [](ApplicationProgramObject* io, uint16_t start, uint8_t count, const uint8_t* data) -> uint8_t 
+            { 
+                memcpy(io->_programVersion, data, 5);
+                return 1;
+            }),
         new CallbackProperty<ApplicationProgramObject>(this, PID_PEI_TYPE, false, PDT_UNSIGNED_CHAR, 1, ReadLv3 | WriteLv0,
```

Comment:
Is there any reason to not use the Dataproperty?

Reply:
I initially tried with the DataProperty but this did not work and I am not sure why. The value was returned correctly during full programming but not when I did partial programming. ETS always stopped with an error "trying to access protected memory" because the property was not returned (count=0). IIRC when I used the CallbackProperty without saving _programVersion to flash I had the same problem. I will test this again and report.